### PR TITLE
Support all services types in the curl subcommand

### PR DESCRIPTION
**Pull request by ean**
Required authentication is chosen based on which service is accessed, and if
you have Auth0 or api-key available.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by ean** [comment]
@mpolden - Please take a look at the latest diff. Based on our discussion yesterday I think this follow the simple wrapper idea you had for the curl subcommand.

@oyving - FYI, this should could you the the functionality you asked for.

File: client/go/cmd/curl.go [hunk 1]
```diff
@@ -60,8 +57,36 @@ $ vespa curl -- -v --data-urlencode "yql=select * from music where album contain
 		if err != nil {
 			return err
 		}
-		c.PrivateKey = privateKeyFile
-		c.Certificate = certificateFile
+		switch curlService {
+		case "deploy":
+			if !vespa.Auth0AccessTokenEnabled() {
```
Comment (mpolden) [inline client/go/cmd/curl.go:62]: A test should be added for this branch.

Comment (ean) [reply client/go/cmd/curl.go]: Added a test for the "target is local" case, would have been nice to have something on the Auth0-part as well, but I see no way to do that without mocking away the entire Auth0-flow, which will make the test useless.

File: client/go/cmd/curl_test.go [hunk 2]
```diff
@@ -18,4 +18,8 @@ func TestCurl(t *testing.T) {
 		filepath.Join(homeDir, "t1.a1.i1", "data-plane-private-key.pem"),
 		filepath.Join(homeDir, "t1.a1.i1", "data-plane-public-cert.pem"))
 	assert.Equal(t, expected, out)
+
+	out, _ = execute(command{homeDir: homeDir, args: []string{"curl", "-s", "deploy", "-n", "/application/v4/tenant/foo"}}, t, httpClient)
+	expected = fmt.Sprintf("curl https://127.0.0.1:19071/application/v4/tenant/foo\n")
```
Comment (mpolden) [inline client/go/cmd/curl_test.go:23]: `fmt.Sprintf` is not needed?

**Review by mpolden — APPROVED** [review]

**Pull request merged**